frontend/packages/ceph-storage-plugin/src/components/ocs-storage-class-form/ocs-thick-provisioner.tsx

```diff
@@ -0,0 +1,43 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+
+import { ProvisionerProps } from '@console/plugin-sdk';
+import { useFlag } from '@console/shared/src/hooks/flag';
+
+import { GUARDED_FEATURES } from '../../features';
+
+import './ocs-storage-class-form.scss';
+
+export const ThickProvision: React.FC<ProvisionerProps> = ({ onParamChange }) => {
+  const { t } = useTranslation();
+  const isThickProvisionSupported = useFlag(GUARDED_FEATURES.THICK_PROVISION);
+
+  const [checked, isChecked] = React.useState(false);
+
+  const setChecked = (value: boolean) => {
+    onParamChange(value.toString());
+    isChecked(value);
+  };
+
+  return (
+    isThickProvisionSupported && (
+      <>
+        <div className="checkbox ocs-storage-class-thickprovision__checkbox">
+          <label>
+            <input
+              type="checkbox"
+              onChange={(event) => setChecked(event.target.checked)}
+              checked={checked}
+            />
+            {t('ceph-storage-plugin~Enable Thick Provisioning.')}
+          </label>
+        </div>
```

Comment:
Use https://github.com/afreen23/console/blob/f92ee559a19a9c220c908967c082b6405342d15c/frontend/public/components/checkbox.tsx#L3 for checkbox